### PR TITLE
feat: Different Oracle Bark Types, default to Never

### PR DESCRIPTION
**Pull request by DisposableCrewmember42**
## About the PR
This PR shuts the Oracle up, as requested by Game Directors [in Discord](https://discord.com/channels/968983104247185448/1283094866670129163/1283100648128254052).

## Why / Balance
The Oracle's barking does not fit the proposed lore and is deemed too obnoxious.

## Technical details
This PR introduces a new enum, OracleBarkType, with three variants:
- Timed: The current behavior
- NewDemand: Bark on NextItem call
- Never: Do not send barks.

It adds a new attribute, BarkType, to the OracleComponent, which is set to Never by default.

I introduced these variants since there was discussion of barking being added with different messages. This way, it could be reintroduced by simply changing the attribute override in the oracle prototype YAML.

## Media

https://github.com/user-attachments/assets/5ba7ab77-a93c-4e3a-b18c-636260bb97f5

## Requirements

- [x] I have read and I am following the [Pull Request Guidelines](https://docs.spacestation14.com/en/general-development/codebase-info/pull-request-guidelines.html). I understand that not doing so may get my pr closed at maintainer’s discretion
- [x] I have added screenshots/videos to this PR showcasing its changes ingame, **or** this PR does not require an ingame showcase

## Breaking changes

**Changelog**

:cl: DisposableCrewmember42
- tweak: Shut the oracle up. Scientists may now research in peace.

### CODE REVIEW

File: Content.Server/Nyanotrasen/Research/Oracle/OracleBarkType.cs
Comment (TadJohnson00) [inline Content.Server/Nyanotrasen/Research/Oracle/OracleBarkType.cs:1]: <sub>I'm just a yamltainer but a newline at the end would be cool</sub>

**Review by TadJohnson00 — COMMENTED** [review]
I support this change, all that's pending now is review from someone who does C#. Thanks for the PR 😎

**Comment by DisposableCrewmember42** [comment]
oh god i fucked up the rebase, this only changes C#, labels for Sprites and YML can be removed

File: Content.Server/Nyanotrasen/Research/Oracle/OracleBarkType.cs [hunk 1]
```diff
@@ -0,0 +1,6 @@
+public enum OracleBarkType {
```
Comment (deltanedas) [inline Content.Server/Nyanotrasen/Research/Oracle/OracleBarkType.cs]: this has no namespace + should just be in the components file + add `: byte`

File: Content.Server/Nyanotrasen/Research/Oracle/OracleComponent.cs [hunk 2]
```diff
@@ -23,6 +23,9 @@ public sealed partial class OracleComponent : Component
     [DataField("barkTime")]
     public TimeSpan BarkTime = TimeSpan.FromMinutes(1);
 
+    [DataField("barkType")]
```
Comment (deltanedas) [inline Content.Server/Nyanotrasen/Research/Oracle/OracleComponent.cs]: remove ("barkType")

**Review by deltanedas — CHANGES_REQUESTED** [review]

**Comment by deltanedas** [comment]
move it into the file, not nested

**Review by deltanedas — APPROVED** [review]

**Review by TadJohnson00 — APPROVED** [review]
Dir. approval. yippee

**Pull request merged**